### PR TITLE
Feature/navigator move to canvas

### LINKED ISSUE #773: move the navigator into the splitpane
Move the navigator out from the left pane next to the left side of the canvas.
The current layout has the code editor in the middle and it cuts the navigator from the canvas.

### PR DESCRIPTION
**Pull request by enidemi**
Fixes #773 

Updating the editor layout a little to have the navigator next to the canvas, change the left pane to open on top of the code editor. 

Changes:
- Rename SplitView
- Remove ToggleInterfaceDesignerLayoutReversed
- Remove all the magical places that toggle focus to the navigator from the other menus of the Left Pane
- The Left Pane is closed by default, the Navigator is open
- Removed the Help Triangle
- The Navigator visibility can be toggled from the Right Menu

Screenshot 2020-12-15 at 18 00 42

### CODE REVIEW

File: editor/src/components/documentation/documentation-components.tsx [hunk 1]
```diff
@@ -70,13 +70,10 @@ export const PrettyKeys = (props: PrettyKeysProps) => {
   return (
     <span>
       {[...props.shortcut].map((keyb, index) => (
-        <span key={(keyb === " ") ? `space-${index}`: keyb} style={{ ...keyStyle }}>
+        <span key={keyb === ' ' ? `space-${index}` : keyb} style={{ ...keyStyle }}>
```
Comment (Rheeseyb) [inline editor/src/components/documentation/documentation-components.tsx:73]: For anyone else that wonders why this change is in here, it looks to have been just a result of Prettier running on the file, likely as a result of latest master being merged into this branch.

File: editor/src/components/canvas/design-panel-root.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            style={{
+              position: 'relative',
+              overflowX: 'hidden',
+              justifyContent: 'stretch',
+              alignItems: 'stretch',
+              flexGrow: 1,
+            }}
+          >
+            <CanvasWrapperComponent {...props} />
+            {navigatorVisible && (
```
Comment (enidemi) [inline editor/src/components/canvas/design-panel-root.tsx:145]: this is where this file actually changed with adding the navigator

**Review by Rheeseyb — APPROVED** [review]

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/right-menu.tsx [hunk 3]
```diff
@@ -116,6 +117,11 @@ export const RightMenu = betterReactMemo('RightMenu', (props: RightMenuProps) =>
     'RightMenu rightMenuSelectedTab',
   )
 
+  const navigatorVisible = useEditorState(
+    (store) => store.editor.navigator.visible,
+    'RightMenu rightMenuSelectedTab',
```
Comment (balazsbajorics) [inline editor/src/components/canvas/right-menu.tsx:122]:
```suggestion
    'RightMenu navigatorVisible',
```

File: editor/src/components/common/actions/index.ts [hunk 4]
```diff
@@ -15,21 +14,22 @@ export type InspectorPanel = 'inspector'
 
 export type CodeEditorPanel = 'uicodeeditor'
 
+export type NavigatorPanel = 'navigatorPane' | 'navigator'
```
Comment (balazsbajorics) [inline editor/src/components/common/actions/index.ts:17]: what is navigator and what is navigatorPane?

File: editor/src/core/performance/performance-regression-tests.spec.tsx [hunk 5]
```diff
@@ -59,8 +59,8 @@ describe('React Render Count Tests - ', () => {
     )
 
     const renderCountAfter = renderResult.getNumberOfRenders()
-    expect(renderCountAfter - renderCountBefore).toBeGreaterThan(475) // if this breaks, GREAT NEWS but update the test please :)
-    expect(renderCountAfter - renderCountBefore).toBeLessThan(485)
+    expect(renderCountAfter - renderCountBefore).toBeGreaterThan(620) // if this breaks, GREAT NEWS but update the test please :)
```
Comment (balazsbajorics) [inline editor/src/core/performance/performance-regression-tests.spec.tsx:62]: this is sub-ideal!

File: editor/src/components/common/actions/index.ts (hunk 4 above)
Comment (enidemi) [reply editor/src/components/common/actions/index.ts:17]: navigatorPane is the fullheight pane which contains the navigator for now, it can be toggled from the right menu

navigator is part of the navigatorPane, it can be closed/open from the arrow toggle on the topright corner of the navigator (it keeps the pane open, but closes the Elements area)

File: editor/src/core/performance/performance-regression-tests.spec.tsx (hunk 5 above)
Comment (enidemi) [reply editor/src/core/performance/performance-regression-tests.spec.tsx:62]: I know.. I'm still trying to run tests locally after seeing all kind of globalThis errors and my direnv had to be reinstalled

Comment (enidemi) [reply editor/src/core/performance/performance-regression-tests.spec.tsx]: fixed

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/common/actions/index.ts (hunk 4 above)
Comment (maltenuhn) [reply editor/src/components/common/actions/index.ts:17]: (minor nitpick / soapbox point: that `Navigator` area inside `<Design>` should be a `pane`. One of the `panels` inside it is the existing navigator. Another (in the future) would be the `DetailNavigator`. Rule: `panes` _contain_ `panels`.
